README.md

```diff
@@ -337,6 +339,190 @@ PING 10.0.0.1 (10.0.0.1) 56(84) bytes of data.
 rtt min/avg/max/mdev = 0.054/0.141/0.342/0.117 ms
 ```
 
+## Testing environment (virtio)
+Below is our testing environment with virtio feature:
+
+<p align="center"><img src="assets/vwifi_virtio_arch.drawio.png" alt="vwifi virtio image" width=60%></p>
+
+vwifi makes use of the `virtio-net` device, Linux tap device, and Linux bridge device to fulfill inter-guest communication.
+In our testing environment, on the host side, we create three `tap` devices and one `bridge` device, then attach each `tap` device to the `bridge` device. And we create three VMs by `Qemu` with the `virtio-net` devices and specify that every `virtio-net` device connects to a `tap` device in the host.
+In VM1, the network interface created by `vwifi` will be in HostAP mode, with the user program `hostapd` running on top of it.
+In VM2 and VM3, we have an STA mode interface and `wpa_supplicant` running on top of it.
+## Build and Run (virtio)
+Below describes how to build a minimal workable VM environment.
+### Build Linux kernel
```

Comment:
Can we skip the section about building Linux kernel? That is, make use of buildroot to prepare everything we need.

Reply:
Building the Linux kernel within Buildroot is a little buggy (I tried many times and failed). I think maybe we can remain the explicit kernel build.